subprojects/core/src/main/java/org/gradle/initialization/DefaultGradleLauncherFactory.java

```diff
@@ -205,8 +214,8 @@ public NestedBuildFactoryImpl(BuildTreeScopeServices buildTreeScopeServices) {
         }
 
         @Override
-        public GradleLauncher nestedInstance(StartParameter startParameter) {
-            return createChildInstance(startParameter, parent, buildTreeScopeServices, ImmutableList.of());
+        public GradleLauncher nestedInstance(StartParameter startParameter, PluginRequests pluginRequests) {
```

Comment:
Instead of `PluginRequests`, this should instead be the "build definition" (or some subset) from the build registry. The "build definition" might provide the `StartParameter` instead, as the `StartParameter` is the current "build definition".

Reply:
Would we do this by having the "build definition" a service that's available in the service registry (like `StartParameter`)?